### PR TITLE
Remove everGotGamepadEvent flag and gamepadconnected / gamepaddisconnected handlers (fix #2769)

### LINKED ISSUE #2769: Daydream controller doesnt work and entering VR mode throws a critical error
**Description:**
Daydream controller doesnt work and entering VR Mode throws a critical error, somehow it works in custom builds and older versions of Aframe.

- A-Frame Version:
Tried various ones, first ofc the latest one 0.5.0 (master/f850c2f),
then some of the last 2 days 0.5.0 (master/9266cfd) and so on,
then I bundled an A-frame version with an older VR Polyfill (0. which PlayCanvas uses in their [WebVR example](https://playcanv.as/p/sAsiDvtC/) because this somehow still works including the Daydream controller,
nothing worked.

- Platform / Device:
**Hardware** ZTE Axon 7 + Google Daydream View
**OS** LineageOS 14.1 running Android 7.1.2 (lineage-14.1-20170612-nightly-axon7-signed) 2017-06-12 10:29:17 
**UA** Mozilla/5.0 (Linux; Android 7.1.2; ZTE A2017U Build/N2G47O) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/59.0.3071.92 Mobile Safari/537.36

Yes its flashed, because ZTE just released a working 7.1 Nougat/Daydream version some days/weeks ago and there was no official version which supported daydream in February. I will also test if the error still occurs with the manufacturer stock version soon, but since it works on PlayCanvas and older A-frame examples i think this _makes no difference_.

- Main error:
Uncaught (in promise) Error: Failed to enter VR mode ('requestPresent'): Presentation request was denied
-> after triggering the VR mode button and thus the Daydream (Pls insert your device) screen never appears.

- Reproducible Code Snippet or URL:
Every **current** example from A-Frame on the mainpage and from private projects using current versions, too i.e. [ground control vr](https://github.com/aframe-capstone/ground-control-vr).
[Codepen](https://codepen.io/intenser/pen/gRMGBq)

- Working examples are:
-- [webvr.yoga](http://webvr.yoga) (working daydream controller)
-- [webvr.playcanvas.com](https://playcanv.as/p/sAsiDvtC/) (even tho it throws other errors and its not completly compareable it still shows the controller works in web with my setup)
-- https://webvr.info/samples/XX-vr-controllers.html?polyfill=1 (this works as intended like in gaze is no controller interaction just by touch you can change the background, then I enter VR mode the daydream window pops up and the controller is able to change the background on click)

- Screens
-- webvr.yoga which works
aframe-webvryoga-test
-- aframe boilerplate with aframe master branch
aframe-boilerplate-test
-- aframe boilerplate with aframe master branch and old polyfill (9.0.25 -> which is the same playcanvas uses atm)
aframe-boilerplate-old-poly-test

### PR DESCRIPTION
**Pull request by dmarcos**

### CODE REVIEW

**Comment by dmarcos** [comment]
@fernandojsg ?

**Comment by fernandojsg** [comment]
@dmarcos it works! :)

**Comment by machenmusik** [comment]
nice catch

**Comment by vincentfretin** [comment]
There is a this.everGotGamepadEvent variable in gearvr-controls.js that seems unused, you may want to remove it too.

File: src/components/daydream-controls.js [hunk 1]
```diff
@@ -93,27 +91,14 @@ module.exports.Component = registerComponent('daydream-controls', {
     this.checkControllerPresentAndSetup(this, GAMEPAD_ID_PREFIX, {hand: this.data.hand});
   },
 
-  onGamepadConnectionEvent: function (evt) {
-    this.everGotGamepadEvent = true;
```
Comment (cvan) [inline src/components/daydream-controls.js]: yeah, per @vincentfretin's comment, there are a few more:

```
src/components/gearvr-controls.js
59:    this.everGotGamepadEvent = false;

tests/components/daydream-controls.test.js
198:      // Reset everGotGamepadEvent so we don't think we've looked before.
199:      delete component.everGotGamepadEvent;

tests/components/gearvr-controls.test.js
211:      // Reset everGotGamepadEvent so we don't think we've looked before.
212:      delete component.everGotGamepadEvent;

tests/components/oculus-touch-controls.test.js
166:      // Reset everGotGamepadEvent so we don't think we've looked before.
167:      delete component.everGotGamepadEvent;

tests/components/vive-controls.test.js
194:      // Reset everGotGamepadEvent to mock not looked before.
195:      delete component.everGotGamepadEvent;

tests/components/windows-motion-controls.test.js
533:      // Reset everGotGamepadEvent so we don't think we've looked before.
534:      delete component.everGotGamepadEvent;
```

**Pull request merged**